php-frontend/src/test/java/org/sonar/php/filters/SuppressWarningFilterTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.filters;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.sonar.php.ParsingTestUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class SuppressWarningFilterTest extends ParsingTestUtils {
```

Comment:
In this class you're using a lot of duplicate Strings which can be made into static variables (`myFile.php`, `SuppressWarnings(\"php:S1234\")`, `php:S1234` for example)

Reply:
I would disagree.
I think there is a lot of value by treating testCode as regular Code. (or relatively close).
I often encountered cases where I had to change strings in testCode where it would have been a lot easier to just change one variable.

I agree, that this is a bit of overkill. For example i would drop the INVALID_PHP_FILE as its really only used one time in the test. For SUPRESS Warnings its a bit of personal opinion to have it as a string concatenation, but i would be fine either way

Reply:
> I think there is a lot of value by treating testCode as regular Code. (or relatively close)
I disagree with this :D

For me, regular code and test code achieve two different goals and for this are written in different ways.
While for normal code I would optimize things, avoid duplication and factorize as much as possible, I consider test code more as a specification of what is expected from the running code. So I'm not afraid of repeating things and be very complete when writing test cases (sometimes probably too much).

Reply:
For the test code is more important to me readability than code duplication. I would like to read the test case without much scrolling here and there to read constant values. See: https://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests

Reply:
I agree with Rudy. In that case, using a constant would decrease the readability of the test case parameters. I would keep the strings otherwise we would have a lot of concats as shown by Rudy's code example. IMO the multiple `+` makes it quite hard to read.